microsoft/iot-cardboard-js
# [Data History Explorer]: Telemetry and Logging

### Summary of changes 🔍 
- Added telemetry for Data History Explorer (for both system and user actions) and logging of actions. Note that I added a separate section for `DataHistoryExplorer` in the `TelemetryConstants` file since it is a independent entity as an orthogonal tool to the 3D scene app (different than Adapters, Builder and Viewer app regions). If you have concerns about it we can discuss more.
- **To-do:** I am going to emit `OpenModal` user action telemetry when I bind the modal to property inspector soon.

![image](https://user-images.githubusercontent.com/45217314/214651119-a04447e9-25d8-462e-999e-7133fd62662d.png)
![image](https://user-images.githubusercontent.com/45217314/214651212-5b57c373-18dc-46cb-a76f-62008bcc839b.png)

Here is some examples:
![image](https://user-images.githubusercontent.com/45217314/214651710-5bbe9082-ad4e-4337-a10f-c13c5d37183b.png)


### Testing 🧪

1. Enable telemetry logging feature in your local storage using `localStorage.setItem('cardboard.debug.telemetryLogging',true)`;
2. Enable "Verbose" to see the telemetry logging in the inspector on the page
3. Go to Components/DataHistoryExplorer/Modal ADT story and see the logs and telemetry

### Checklist ✔️
